File: ClinicSystem/src/com/example/clinicsystem/models/classes/M_DentalReport.java
package com.example.clinicsystem.models.classes;

import com.example.clinicsystem.models.interfaces.IAddDetail;

public class M_DentalReport implements IAddDetail {
    private int reportID, patientID, dentistID, cavityCount, fillingDecayCount, appointmentID;
    private String date, gumStatus, xray, consentProof;
    private boolean isPatientConsent;

    @Override
    public void addDetail(M_Detail detail) {
        detail.parentID = this.reportID;
    }

    public M_DentalReport(int reportID, int patientID, int dentistID, int cavityCount, int fillingDecayCount, int appointmentID, String date, String gumStatus, String xray, String consentProof, boolean isPatientConsent) {
        this.reportID = reportID;
        this.patientID = patientID;
        this.dentistID = dentistID;
        this.cavityCount = cavityCount;
        this.fillingDecayCount = fillingDecayCount;
        this.appointmentID = appointmentID;
        this.date = date;
        this.gumStatus = gumStatus;
        this.xray = xray;
        this.consentProof = consentProof;
        this.isPatientConsent = isPatientConsent;
    }

    public M_DentalReport() {
    }

    public int getReportID() {
        return reportID;
    }

    public void setReportID(int reportID) {
        this.reportID = reportID;
    }

    public int getPatientID() {
        return patientID;
    }

    public void setPatientID(int patientID) {
        this.patientID = patientID;
    }

    public int getDentistID() {
        return dentistID;
    }

    public void setDentistID(int dentistID) {
        this.dentistID = dentistID;
    }

    public int getCavityCount() {
        return cavityCount;
    }

    public void setCavityCount(int cavityCount) {
        this.cavityCount = cavityCount;
    }

    public int getFillingDecayCount() {
        return fillingDecayCount;
    }

    public void setFillingDecayCount(int fillingDecayCount) {
        this.fillingDecayCount = fillingDecayCount;
    }

    public int getAppointmentID() {
        return appointmentID;
    }

    public void setAppointmentID(int appointmentID) {
        this.appointmentID = appointmentID;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getGumStatus() {
        return gumStatus;
    }

    public void setGumStatus(String gumStatus) {
        this.gumStatus = gumStatus;
    }

    public String getXray() {
        return xray;
    }

    public void setXray(String xray) {
        this.xray = xray;
    }

    public String getConsentProof() {
        return consentProof;
    }

    public void setConsentProof(String consentProof) {
        this.consentProof = consentProof;
    }

    public boolean isPatientConsent() {
        return isPatientConsent;
    }

    public void setPatientConsent(boolean patientConsent) {
        isPatientConsent = patientConsent;
    }
}

File: clinic/M_DentalReport.php
<?php
include_once "IAddDetail.php";

class M_DentalReport implements IAddDetail {
    private $reportID, $patientID, $dentistID, $cavityCount, $fillingDecayCount, $appointmentID;
    private $ddate, $gumStatus, $xray, $consentProof;
    private $isPatientConsent;

    public function addDetail($detail) {
        $detail->parentID = $this->reportID;
    }

    public function __construct($reportID, $patientID, $dentistID, $cavityCount, $fillingDecayCount, $appointmentID, $ddate, $gumStatus, $xray, $consentProof, $isPatientConsent) {
        $this->reportID = 			$reportID;
        $this->patientID = 			$patientID;
        $this->dentistID = 			$dentistID;
        $this->cavityCount = 		$cavityCount;
        $this->fillingDecayCount =  $fillingDecayCount;
        $this->appointmentID = 		$appointmentID;
        $this->ddate = 				$ddate;
        $this->gumStatus = 			$gumStatus;
        $this->xray = 				$xray;
        $this->consentProof = 		$consentProof;
        $this->isPatientConsent = 	$isPatientConsent;
    }

    public function getReportID() {
        return $this->reportID;
    }

    public function setReportID($reportID) {
        $this->reportID = reportID;
    }

    public function getPatientID() {
        return $this->patientID;
    }

    public function setPatientID($patientID) {
        $this->patientID = $patientID;
    }

    public function getDentistID() {
        return $this->dentistID;
    }

    public function setDentistID($dentistID) {
        $this->dentistID = $dentistID;
    }

    public function getCavityCount() {
        return $this->cavityCount;
    }

    public function setCavityCount($cavityCount) {
        $this->cavityCount = $cavityCount;
    }

    public function getFillingDecayCount() {
        return $this->fillingDecayCount;
    }

    public function setFillingDecayCount($fillingDecayCount) {
        $this->fillingDecayCount = $fillingDecayCount;
    }

    public function getAppointmentID() {
        return $this->appointmentID;
    }

    public function setAppointmentID($appointmentID) {
        $this->appointmentID = $appointmentID;
    }

    public function getdDate() {
        return $this->ddate;
    }

    public function setDate($ddate) {
        $this->ddate = $ddate;
    }

    public function getGumStatus() {
        return $this->gumStatus;
    }

    public function setGumStatus($gumStatus) {
        $this->gumStatus = $gumStatus;
    }

    public function getXray() {
        return $this->xray;
    }

    public function setXray($xray) {
        $this->xray = $xray;
    }

    public function getConsentProof() {
        return $this->consentProof;
    }

    public function setConsentProof($consentProof) {
        $this->consentProof = $consentProof;
    }

    public function isPatientConsent() {
        return $this->isPatientConsent;
    }

    public function setPatientConsent($patientConsent) {
        $this->isPatientConsent = $patientConsent;
    }
}
?>